app/models/assessment_user_datum.rb

```diff
@@ -155,6 +155,8 @@ def invalidate_cgdubs_for_assessments_after
       end
 
       Rails.cache.delete course_user_datum.ggl_cache_key
+
+      course_user_datum.update_cud_gdu_watchlist_instances
```

Comment:
hi this is just because I'm not very clear on the caching is working here, but could you explain how this part works?

Reply:
This added line doesn't have anything to do with the caching part. I added this line inside of the transaction with the lock on just to maintain consistency. I believe whenever cgdubs are invalidated, we need to update the watchlist instances (since anything could happen to grace days used) so I thought it would be a good idea to immediately ask CUD to update after invalidating.